### PR TITLE
Can we start using nix 2.4?

### PR DESCRIPTION
**Pull request by zupo**

### CODE REVIEW

**Comment by NorfairKing** [comment]
I can't repro the issue locally so I'm hoping a bump can make CI pass.

**Comment by NorfairKing** [comment]
@zupo I think may have just been flaky.

**Comment by NorfairKing** [comment]
All but the unstable passes now

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -22,14 +22,14 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        channel: [ "nixos-21.11", "nixos-21.05", "nixos-20.09", "nixos-unstable" ]
+        channel: [ "nixos-21.11", "nixos-21.05", "nixos-20.09", "nixos-unstable"]
```
Comment (zupo) [inline .github/workflows/push.yml:25]: Why the whitespace change here? Is this the best-practice format?

**Comment by zupo** [comment]
> @zupo I think may have just been flaky.

Maybe the fix got pushed in the recent cachix action or niv updates. We just got lucky, someone else fixed it for us? :)

That said, we need unstable to work too before we can merge it, otherwise, we don't get any new unstable images, and we need them.

File: .github/workflows/push.yml (hunk 1 above)
Comment (NorfairKing) [reply .github/workflows/push.yml:25]: If I did this then it was by accent. Will undo when I squash this branch :)

**Comment by zupo** [comment]
We're not going to do this until nixpkgs stable channel upgrades to nix 2.4.

**Pull request closed without merging**